internal/proto/plugin/v1/host_plugin_service.proto

```diff
@@ -10,8 +10,13 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/plugin;plugin";
 
 // HostPluginService describes the service for host plugins.
 service HostPluginService {
-  // OnCreateCatalog is a hook that runs when a host catalog is
-  // created.
+  // NormalizeCatalogData is a hook that passes attributes to the plugin and
+  // allows those values to be normalized prior to creating or updating those
+  // values in the host catalog data.
+  rpc NormalizeCatalogData(NormalizeCatalogDataRequest) returns (NormalizeCatalogDataResponse);
```

Comment:
Can we make it explicit which methods `NormalizeCatalogData` is called before? For example:
```
NormalizeCatalogData is called before:
* OnCreateCatalog
* OnUpdateCatalog
```

Reply:
Done